hypothesis-python/tests/cover/test_regressions.py

```diff
@@ -77,3 +77,27 @@ def test_foo_spec(bar, inp):
     test_foo_spec(Bar())
     test_foo_spec(Mock(Bar))
     test_foo_spec(Mock())
+
+
+def test_regression_issue_1230():
+    strategy = st.builds(
+        lambda x, y: dict((list(x.items()) + list(y.items()))),
+        st.fixed_dictionaries({'0': st.text()}),
+        st.builds(
+            lambda dictionary, keys: {key: dictionary[key] for key in keys},
+            st.fixed_dictionaries({
+                '1': st.lists(elements=st.sampled_from(['a']))
+            }),
+            st.sets(elements=st.sampled_from(['1']))
+        )
+    )
+
+    @seed(81581571036124932593143257836081491416)
+    @settings(database=None)
+    @given(strategy)
+    def test_false_is_false(params):
+        assume(params.get('0') not in ('', '\x00'))
+        raise ValueError()
+
+    with pytest.raises(ValueError):
```

Comment:
nit: since ValueError is an exception that can be thrown in stdlib code, it’s presumably possible to hit a (different) error in Hypothesis core that throws a ValueError, and then this test passes even though it shouldn’t.

If I was writing this from scratch, I’d define a TestSentinel exception that only lives in the test suite, and throw/catch that here – so we can never confuse an error in core with the expected exception class.

In practice, I think that’s very unlikely, so it’s not a “must address” comment.

Reply:
Yeah, we should probably do something like that more generally across our code, but I'm not sure I'm sufficiently motivated right now. :-)